Assets/Scripts/Interaction Scripts/ThirdPersonView.cs

```diff
@@ -0,0 +1,450 @@
+                        controllerState = ControllerState.IDLE;
+                        RightPointer.enabled = true;
+                        RightPointerRenderer.enabled = true;
+                        RightUIPointer.enabled = true;
+                        /// TODO: make waypoint disappear
+                        break;
+                    }
+                    controllerState = ControllerState.IDLE;
+                    break;
+                }
+
+            }
+
+        }
+
+
+        private void ScaleWorld()
+        {
+            // Get the scaling factor, and adjust its size.
+            float ScalingFactor = 1.0f + 0.2f * controllerInput.GetScalingFactor();
+            Vector3 ScalingVector = Vector3.Scale(World.transform.localScale, new Vector3(ScalingFactor, ScalingFactor, ScalingFactor));
+
+            //Checking Scaling Bounds
+            if (ScalingVector.sqrMagnitude > WorldScaleMin.sqrMagnitude && ScalingVector.sqrMagnitude < WorldScaleMax.sqrMagnitude)
+            {
+                // FIXME: Jank. and comments.
+                Vector3 A = World.transform.position;
+                Vector3 B = Pivot.transform.position;
+                B.y = A.y;
+
+                Vector3 startScale = World.transform.localScale;
+                Vector3 endScale = World.transform.localScale * ScalingFactor;
+
+                Vector3 C = A - B; // diff from object pivot to desired pivot/origin
+
+                // calc final position post-scale
+                Vector3 FinalPosition = (C * ScalingFactor) + B;
+
+                // finally, actually perform the scale/translation
+                World.transform.localScale = endScale;
+                World.transform.position = FinalPosition;
+            }
+
+        }
+
+
+        // Rotate the world based off of the right thumbstick
+        private void RotateWorld()
+        {
+            float angle;
+            if (direction == Direction.REGULAR)
+            {
+                angle = -controllerInput.GetRightThumbDelta().x * rotationalSpeed * 360 * Time.fixedDeltaTime;
+            }
+            else
+            {
+                angle = controllerInput.GetRightThumbDelta().x * rotationalSpeed * 360 * Time.fixedDeltaTime;
+            }
+            World.transform.RotateAround(Pivot.transform.position, Vector3.up, angle);
+
+            // Peru: 5/28/2020 : Point Cloud Rotate
+            // FIXME: Is declaring here correct?
+            GameObject pointCloud = GameObject.Find("PointCloud");
+
+            if (pointCloud)
+            {
+                pointCloud.transform.RotateAround(Pivot.transform.position, Vector3.up, angle);
+            }
+        }
+
+
+        private void MoveWorld()
+        {
+            // Negative values are here to make moving around look natural.
+            // Without occlusion this looks mediocre because it seems like the map is being moved in the wrong direction.
+            float moveX = controllerInput.GetLeftThumbDelta().x;
```

Comment:
are these given in the range[0, 1] or [-1, 1]?

Reply:
I take it from the -= that its [0,1] ?

Reply:
Yes, it's [0, 1].

Reply:
It's by how much we move the thumbstick away from the center, in any direction.